sdk/python/kfp/dsl/_component_bridge.py

```diff
@@ -46,20 +46,36 @@ def _create_container_op_from_component_and_arguments(
 
     old_warn_value = dsl.ContainerOp._DISABLE_REUSABLE_COMPONENT_WARNING
     dsl.ContainerOp._DISABLE_REUSABLE_COMPONENT_WARNING = True
+
+    # Merge output_paths and output_uris to get the file_outputs.
+    file_outputs = resolved_cmd.output_paths or {}
```

Comment:
> Should this be a copy of output_paths

I think to keep everything in consistent I slightly prefer modify in-place. Though I cannot find any usage of `resolved_cmd.output_paths` below.

Changed to modify in place always.

Reply:
If I read the code correctly, `resolved_cmd.output_paths` is a dictionary and if it's not empty, then `file_output` points to the same dictionary object, right?

so `file_output[key] = value` is actually updating(augmenting) `resolved_cmd.output_paths`

Reply:
> so file_output[key] = value is actually updating(augmenting) resolved_cmd.output_paths

It does not seems the case. I did some experimental by printing the result 

```python
    # Merge output_paths and output_uris to get the file_outputs.
    file_outputs = resolved_cmd.output_paths or {}
    for name, output_uri in resolved_cmd.output_uris.items():
    file_outputs[name] = output_uri
    print('Now file_outputs is:')
    print(file_outputs)
    print('... and resolved_cmd.output_paths is')
    print(resolved_cmd.output_paths)
```

And for `test_uri_artifact_passing` in `compiler_tests.py` it generates the following result:
```
Now file_outputs is:
{'output_gcs_path': '{{kfp.output_dir}}/{{workflow.uid}}/{{pod.name}}/output_gcs_path'}
... and resolved_cmd.output_paths is
OrderedDict()
```

Reply:
>... and resolved_cmd.output_paths is
>OrderedDict()

This result shows that in your experiment `resolved_cmd.output_paths` was initially empty. So `file_outputs = resolved_cmd.output_paths or {}` resolved to `file_outputs = {}` rather than `file_outputs = resolved_cmd.output_path`. If `resolved_cmd.output_paths` weren't empty, then `file_outputs` and `resolved_cmd.output_paths` would point to the same object in memory.

So if you try this code:
```
import collections
dict1 = collections.OrderedDict({'a':1})
dict2 = dict1
dict2['b'] = 2
print(dict1)
```
It will print out:
```
OrderedDict([('a', 1), ('b', 2)])
```
(Same behavior for regular dict).

To avoid this, you can do `file_outputs = collections.OrderedDict(resolved_cmd.output_paths or {})` or  `file_outputs = resolved_cmd.output_paths.copy()` if you can assure `resolved_cmd.output_path` would never be `None`.

Reply:
Thanks for the explanation! Makes sense to me.